src/main/java/com/google/sps/servlets/StudyServlet.java

```diff
@@ -0,0 +1,169 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.gson.Gson;
+
+import com.google.sps.data.Card;
+import com.google.sps.tool.WordSearch;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Collections;
+
+/**
+ * Handles the study mode questions and responses by adjusting and
+ * sorting a card's based on their familarity score and last time seen.
+ */
+@WebServlet("/study")
+public class StudyServlet extends HttpServlet {
+  public class QuizCard {
+    private String quizWord;
+    private String cardKey;
+    private ArrayList<String> possibleResponses = new ArrayList<>();
+    private String correctAnswer;
+
+    public QuizCard(Card card) {
+      this.quizWord = card.getTextTranslated();
+      this.cardKey = card.getCardKey();
+      this.correctAnswer = card.getRawText().toLowerCase();
+      this.possibleResponses = WordSearch.generateWordOptions(this.correctAnswer);
+    }
+  }
+
+  class SortByFamilarity implements Comparator<Card> {
```

Comment:
you can use a lambda expression. Since you are utilizing a List called userCards, you can do something like 
userCards.sort(
                 (Card c1, Card c2) -> Double.compare(c1.getFamiliarityScore(), c2.getFamiliarityScore()));

Reply:
Yeah this seems like a great way to clean up my code

Reply:
done.